expression/function_traits.go

```diff
@@ -54,6 +54,9 @@ var unFoldableFunctions = map[string]struct{}{
 // are in child scope of an outer function, and the outer function is recursively folding its children.
 var DisableFoldFunctions = map[string]struct{}{
 	ast.Benchmark: {},
+	ast.If:        {},
+	ast.Ifnull:    {},
```

Comment:
disabling the constant folding optimizations for these functions may bring performance regressions for existing applications. could you elaborate on the reason for disabling these functions?

Reply:
Could we use the similar fallback policy on constant folding? @fzhedu

Reply:
cc @zz-jason from an oncall, the additional warnings/erros from unknown branches cause a normal `update` statement failed. for example: if(1,1,1/0) should not return errors/warnings, but acutall returns due to directly folding constrants. In contrast to correctness, reducing litttle performance is ok.

Reply:
Can we fix the constant folding algorithm to not return an error or warning for cases like `if(1,1,1/0)`? Can we fix it by short circuit evaluation in the constant evaluation algorithm?

Reply:
It's hard to see how much the performance is impacted. in some cases, folding `if` or `ifnull()` statement can let some filter to be pushed to tikv or enables us to utilize index range scan over table full scan.

Reply:
I try it in #19675